apps/omg_eth/lib/omg_eth/release_tasks/set_contract.ex

```diff
@@ -34,89 +36,71 @@ defmodule OMG.Eth.ReleaseTasks.SetContract do
-          contract_addresses,
-          min_exit_period_seconds,
-          contract_semver,
-          network
-        )
-
-      {_, via_env} when is_binary(via_env) ->
-        :ok = apply_static_settings(via_env)
-
-      _ ->
-        exit(@error)
-    end
-  end
-
-  defp apply_static_settings(network) do
-    network = get_network(network)
-
-    txhash_contract = get_env("TXHASH_CONTRACT")
-    authority_address = get_env("AUTHORITY_ADDRESS")
-    env_contract_address_plasma_framework = get_env("CONTRACT_ADDRESS_PLASMA_FRAMEWORK")
-    env_contract_address_eth_vault = get_env("CONTRACT_ADDRESS_ETH_VAULT")
-    env_contract_address_erc20_vault = get_env("CONTRACT_ADDRESS_ERC20_VAULT")
-    env_contract_address_payment_exit_game = get_env("CONTRACT_ADDRESS_PAYMENT_EXIT_GAME")
+    network = get_network(via_env)
+
+    {txhash_contract, authority_address, plasma_framework} =
+      case exchanger do
+        exchanger when is_binary(exchanger) ->
+          body =
+            try do
+              {:ok, %{body: body}} = HTTPoison.get(exchanger)
+              body
+            rescue
+              reason -> exit("CONTRACT_EXCHANGER_URL #{exchanger} is not reachable because of #{inspect(reason)}")
+            end
+
+          %{
+            authority_address: authority_address,
+            plasma_framework: plasma_framework,
+            plasma_framework_tx_hash: txhash_contract
+          } = Jason.decode!(body, keys: :atoms!)
+
+          {txhash_contract, authority_address, plasma_framework}
+
+        _ ->
+          txhash_contract = get_env("TXHASH_CONTRACT")
+          authority_address = get_env("AUTHORITY_ADDRESS")
+          plasma_framework = get_env("CONTRACT_ADDRESS_PLASMA_FRAMEWORK")
+          {txhash_contract, authority_address, plasma_framework}
+      end
+
+    # get all the data from external sources
+    {payment_exit_game, eth_vault, erc20_vault, min_exit_period_seconds, contract_semver, child_block_interval} =
+      get_external_data(plasma_framework)
 
     contract_addresses = %{
-      plasma_framework: env_contract_address_plasma_framework,
-      eth_vault: env_contract_address_eth_vault,
-      erc20_vault: env_contract_address_erc20_vault,
-      payment_exit_game: env_contract_address_payment_exit_game
+      plasma_framework: plasma_framework,
+      eth_vault: eth_vault,
+      erc20_vault: erc20_vault,
+      payment_exit_game: payment_exit_game
     }
 
-    min_exit_period_seconds = get_min_exit_period(env_contract_address_plasma_framework)
-    contract_semver = get_contract_semver(env_contract_address_plasma_framework)
-
     update_configuration(
       txhash_contract,
       authority_address,
       contract_addresses,
       min_exit_period_seconds,
       contract_semver,
-      network
+      network,
+      child_block_interval
     )
+
+    (logger_apps ++ ethereumex_apps) |> Enum.reverse() |> Enum.each(&Application.stop/1)
+  end
+
+  defp get_external_data(plasma_framework) do
```

Comment:
SetContract release task needed a refactor + I've expanded it to retrieve all contract data from plasma contract